apps/marketplace/actions/appActions.js

```diff
@@ -72,12 +72,16 @@ export const fetchAuth = () => dispatch => {
   })
 }
 
-export const login = data => dispatch => {
+export const login = data => (dispatch, getState) => {
   dispatch(sendingRequest(true))
   dispatch(setAuthFrameworkError(false))
   dmapi({
     method: 'post',
     url: '/login',
+    headers: {
+      'X-CSRFToken': getState().app.csrfToken,
```

Comment:
There is some talk in the React community that using `getState` in a redux action is a bad pattern, breaking the 1 way flow of data, among other concerns. On the other hand, [some wisdom suggests it serves a purpose](https://blog.isquaredsoftware.com/2017/01/idiomatic-redux-thoughts-on-thunks-sagas-abstraction-and-reusability/) and I think in this case adding a global state value like a CSRF token to each action that needs it makes sense. The alternative approach was to add the token as a hidden field in the render of all relevant components/pages and is still possible if there is strong objections to the `getState` approach.